detection_rules/rule.py

```diff
@@ -46,6 +49,42 @@
 }
 
 
+@dataclass
+class DictRule:
+    """Simple object wrapper for raw rule dicts."""
+
+    contents: dict
+    path: Optional[Path] = None
+
+    @property
+    def metadata(self) -> dict:
+        """Metadata portion of TOML file rule."""
+        return self.contents.get('metadata', {})
+
+    @property
+    def data(self) -> dict:
+        """Rule portion of TOML file rule."""
+        return self.contents.get('data') or self.contents
+
+    @property
+    def id(self) -> str:
+        """Get the rule ID."""
+        return self.data['rule_id']
+
+    @property
+    def name(self) -> str:
+        """Get the rule name."""
+        return self.data['name']
+
+    def __hash__(self):
```

Comment:
```suggestion
    def __hash__(self) -> int:
```

Nit I think some of the functions are missing return types. Also for the functions that return none, could we add an explicit `-> None` to match the styling of the other None returns in the repo? Thanks!